### ISSUE TITLE
Missing branch coverage in typescript async method

### ISSUE THREAD

**Issue by RoniSegal** [istanbuljs/istanbuljs#291]
I have the bellow method written in typescript and transpiled to es5:

![image](https://user-images.githubusercontent.com/26412921/52623544-3d111a00-2eb5-11e9-9654-285d7a128233.png).

The transpiled code is:
```
SchedulerController.prototype.pollOnce = function () {
        return __awaiter(this, void 0, void 0, function () {
            var logger, scheduledEvent, err_1;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        logger = this.logger.withMethod("pollOnce");
                        _a.label = 1;
                    case 1:
                        _a.trys.push([1, 5, , 6]);
                        return [4 /*yield*/, this.getNextScheduledEvent(logger)];
                    case 2:
                        scheduledEvent = _a.sent();
                        if (!scheduledEvent) return [3 /*break*/, 4];
                        return [4 /*yield*/, this.runJob(scheduledEvent)];
                    case 3:
                        _a.sent();
                        _a.label = 4;
                    case 4: return [3 /*break*/, 6];
                    case 5:
                        err_1 = _a.sent();
                        logger.error("Error in iteration", err_1);
                        return [3 /*break*/, 6];
                    case 6: return [2 /*return*/];
                }
            });
        });
    };
```
The source map data for
```
 if (!scheduledEvent) return [3 /*break*/, 4];
```
points to line 126

### Expected Behavior
line 126 should considered as branch

### Observed Behavior
line 126 considered as statement

**Comment by coreyfarrell** [comment]
From what you're showing this line should be considered both a branch `if (!scheduledEvent)` and a statement `return [3 /*break*/, 4];`.  To triage this issue we will need to see a minimal reproduction.  If you create a minimal demo repository showing the build / test process it will help us help you.

**Comment by RoniSegal** [comment]
I created [this](https://github.com/RoniSegal/missing-branch-coverage) repo, you can see that in file 'lib/scheduler-controller.ts' 0 branches reported,  but actually  it contains 4 'if' and 1 'else'

**Comment by RoniSegal** [comment]
Any updates on this?

**Comment by coreyfarrell** [comment]
@JaKXz https://github.com/RoniSegal/missing-branch-coverage was provided - the link is easy to miss in @RoniSegal's comment from two days ago.

**Comment by RoniSegal** [comment]
@JaKXz sorry for that, I updated the comment and now its more clear

**Comment by RoniSegal** [comment]
any update on this?

**Comment by coreyfarrell** [comment]
@RoniSegal I just tried your repo:
```
git clone https://github.com/RoniSegal/missing-branch-coverage
cd missing-branch-coverage
npm install
npm test
```
Got the following error:
```
> nyc --reporter=html mocha tsOutputs/test

Warning: Could not find any test files matching pattern: tsOutputs/test
No test files found
npm ERR! Test failed.  See above for more details.
```

Please fix this error, once the test is working someone can take a look.  Keep in mind this issue will require someone with knowledge of both istanbul and TypeScript, so it may take longer to get help.  Any updates will be posted here.

One other suggestion, take a look at [@istanbuljs/nyc-config-typescript](https://www.npmjs.com/package/@istanbuljs/nyc-config-typescript).

**Comment by RoniSegal** [comment]
@coreyfarrell  Just updated my repo, it missed npm run build

**Comment by coreyfarrell** [comment]
I'm able to reproduce this issue now.  I checked the the `if (event === "failed")` branch.  `tsc` translated this to `if (event == "failed")` at line 70.  In the raw coverage data I was able to see that we saw the branch at line 70, we recorded that the tests hit this branch.  But then it was not displayed in the final report.  I suspect this is showing that something went wrong with source-map remapping, so this is likely a bug in `istanbul-lib-source-maps`.

**Comment by RoniSegal** [comment]
@coreyfarrell thanks for looking into it, should i open a new issue and point it to ```istanbul-lib-source-map``?

**Comment by coreyfarrell** [comment]
I'm pretty sure we can transfer this issue (@JaKXz?).

**Comment by RoniSegal** [comment]
@coreyfarrell Seems that ```istanbul-lib-source-maps``` is part of istanbuljs repo (under ```packages```),
Is it still should be transfered>

**Comment by coreyfarrell** [comment]
@RoniSegal that is correct, I got mixed up for some reason thought this was reported against nyc.

**Comment by RoniSegal** [comment]
Any updates on this?
Thanks

**Comment by coreyfarrell** [comment]
@RoniSegal Sorry there is not, source maps issues are very difficult.  If any updates are available they will be posted here.

**Comment by nadavye** [comment]
Hi Guys,
Any update on this one?
